src/app/data/services/local_storage_service.ts

```diff
@@ -0,0 +1,30 @@
+///<reference path="../models/content_block.ts"/>
+///<reference path="ajax_content_blocks_hash_serializer.ts"/>
+
+class LocalStorageService {
+  static EXPIRATIONTIME = 2592000000; //ms (30 days)
+  private expiryDate:{[key: string]: number} = {};
+  private serializer = new AjaxContentBlocksHashSerializer();
+
+  fetch(key:string):AjaxContentBlocksHash {
+    return this.serializer.parse(localStorage.getItem(key));
+  }
+
+  put(key:string, hash:AjaxContentBlocksHash) {
+    if (this.isValid(key)) return;
```

Comment:
> then we have to switch two statements..
is 'if' not preferred over 'if !' ?

@cyrilkyburz agree. but then let's find a better name for isValid. what do you think about ```skipStoring```?